### ISSUE TITLE
page.goBack or page.goForward would crash the test?

### ISSUE THREAD

**Issue by CommanderXL** [puppeteer/puppeteer#1166]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 0.13.0-alpha
* Platform / OS version: OS EI 10.11
* URLs (if applicable):

**What steps will reproduce the problem?**

Well. I try to test the method of `goBack` and `goForward`.But it appears something unexpected, just like the code as follow:

```javascript
const puppeteer = require('puppeteer')

;(async () => {
  const browser = await puppeteer.launch({
    headless: false
  })
  const page = await browser.newPage()
  await page.goto('https://www.baidu.com')
  await new Promise(resolve => setTimeout(resolve, 2000))
  await page.type('.s_ipt', '测试', {
    delay: 500
  })
  await page.click('#su')
  await new Promise(resolve => setTimeout(resolve, 2000))
  await page.goBack()
 
  console.log('back')

  await new Promise(resolve => setTimeout(resolve, 2000))
  await page.goForward()

  console.log('forward')

  await browser.close()
})()
```

**What is the expected result?**

The expected result is the script has run , the terminal will print 'back' after the page go back and print 'forward' after the go forward, and the browser was closed in the end.

**What happens instead?**

But in the terminal, there is no 'back' or 'forward',and the stdout appear the message that 

> (node:39404) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Navigation Timeout Exceeded: 30000ms exceeded
(node:39404) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

and the browser wasn't closed. And the terminal and puppeteer didn't exit.

---- 

### Update:

```javascript
async waitForNavigation() {
    this._requestIds = new Set();

    this._eventListeners = [];

    const navigationPromises = [];
    if (this._timeout) {
      const watchdog = new Promise(fulfill => this._maximumTimer = setTimeout(fulfill, this._timeout))
          .then(() => 'Navigation Timeout Exceeded: ' + this._timeout + 'ms exceeded');
      navigationPromises.push(watchdog);
    }

    if (!this._ignoreHTTPSErrors) {
      const certificateError = new Promise(fulfill => {
        this._eventListeners.push(helper.addEventListener(this._client, 'Security.certificateError', fulfill));
      }).then(error => 'SSL Certificate error: ' + error.errorType);
      navigationPromises.push(certificateError);
    }

    if (this._waitUntil === 'load') {
      const loadEventFired = new Promise(fulfill => {
        this._eventListeners.push(helper.addEventListener(this._client, 'Page.loadEventFired', fulfill));
      }).then(() => null);
      navigationPromises.push(loadEventFired);
    } else {
      this._eventListeners.push(...[
        helper.addEventListener(this._client, 'Network.requestWillBeSent', this._onLoadingStarted.bind(this)),
        helper.addEventListener(this._client, 'Network.loadingFinished', this._onLoadingCompleted.bind(this)),
        helper.addEventListener(this._client, 'Network.loadingFailed', this._onLoadingCompleted.bind(this)),
        helper.addEventListener(this._client, 'Network.webSocketCreated', this._onLoadingStarted.bind(this)),
        helper.addEventListener(this._client, 'Network.webSocketClosed', this._onLoadingCompleted.bind(this)),
      ]);
      const networkIdle = new Promise(fulfill => this._networkIdleCallback = fulfill).then(() => null);
      navigationPromises.push(networkIdle);
    }

    const error = await Promise.race(navigationPromises);
    this._cleanup();
    return error ? new Error(error) : null;
  }
```

In the `NavigatorWatcher.js`.There are three promises in the navigationPromises of my test code.And the the first watchdog always execute after the timeout option. And the other two promise will never be resolved.So the error always `Navigation Timeout Exceeded: (timeout)ms exceeded'` and the error will be thrown out.

**Comment by vsemozhetbyt** [comment]
FWIW, the example code works for me as expected with puppeteer v0.13.0-alpha  in Windows 7 x64.

**Comment by CommanderXL** [comment]
@vsemozhetbyt Thanks for your answer. But I use my colleagues mac OS to run the demo. It occurred the same problem. Could you please use mac to run this demo? Thanks.

**Comment by vsemozhetbyt** [comment]
Unfortunately, I cannot, sorry(

**Comment by aslushnikov** [comment]
I can repro this on puppeteer v1.0.0 on Mac OS X. Going forward hangs.

**Comment by yanivefraim** [comment]
Just tested this on Mac using Puppeteer V1.0.0 and it V1.2.0 - everything was working perfectly for me...

**Comment by aslushnikov** [comment]
@yanivefraim this still doesn't work for me on tip-of-tree: I never get to see the "forward" in the console.
A quick debugging revealed that we don't receive any lifecycle events after navigating forward.

**Comment by aslushnikov** [comment]
I believe this got fixed together with #257. Works fine for me on puppeteer 1.4.0.
